src/components/TextEditor.js

```diff
@@ -0,0 +1,232 @@
+import React from 'react';
+import {Editor, EditorState, getDefaultKeyBinding, RichUtils, convertToRaw, convertFromRaw} from 'draft-js';
+import '../css/RichEditor.css'
+import '../../node_modules/draft-js/dist/Draft.css'
+
+var DESCRIPTIONRAW;
+var STATEHASCHANGED = false;
+var STOREDSTATE = EditorState.createEmpty();
+
+class TextEditor extends React.Component {
+    constructor(props) {
+      super(props);
+      this.state = {editorState : STOREDSTATE};
+      this.focus = () => this.refs.editor.focus();
+      this.onChange = (editorState) => {
+        this.setState({editorState});
+        DESCRIPTIONRAW = convertToRaw(editorState.getCurrentContent())
+        STATEHASCHANGED = true;
+      }
+      this.handleKeyCommand = this._handleKeyCommand.bind(this);
+      this.mapKeyToEditorCommand = this._mapKeyToEditorCommand.bind(this);
+      this.toggleBlockType = this._toggleBlockType.bind(this);
+      this.toggleInlineStyle = this._toggleInlineStyle.bind(this);
+    }
+
+
+    _handleKeyCommand(command, editorState) {
+      const newState = RichUtils.handleKeyCommand(editorState, command);
+      if (newState) {
+        this.onChange(newState);
+        return true;
+      }
+      return false;
+    }
+
+    _mapKeyToEditorCommand(e) {
+      if (e.keyCode === 9 /* TAB */) {
+        const newEditorState = RichUtils.onTab(
+          e,
+          this.state.editorState,
+          4, /* maxDepth */
+        );
+        if (newEditorState !== this.state.editorState) {
+          this.onChange(newEditorState);
+        }
+        return;
+      }
+      return getDefaultKeyBinding(e);
+    }
+
+    _toggleBlockType(blockType) {
+      this.onChange(
+        RichUtils.toggleBlockType(
+          this.state.editorState,
+          blockType
+        )
+      );
+    }
+
+    _toggleInlineStyle(inlineStyle) {
+      this.onChange(
+        RichUtils.toggleInlineStyle(
+          this.state.editorState,
+          inlineStyle
+        )
+      );
+    }
+
+    render() {
+      if(!STOREDSTATE){
+        this.setState({STOREDSTATE});        
```

Comment:
doesn't this simply set empty stated? If so, why is this needed?

Reply:
the set has to be set to either the created empty from Editor or from existing. otherwise the component doesn't work